File: frontend/src/features/table/types/si.ts
import type { ILocationDTO } from '../modules/locations/types/location'
import type { IVerificationDTO } from '../modules/verification/types/verification'
import type { IInstrumentDTO } from './instrument'
import type { ISort, IFilter, ISearch } from './params'

export interface ISI {
	id: string
	position: number
	name: string
	dateOfReceipt: string
	type: string
	factoryNumber: string
	measurementLimits: string
	accuracy: string
	stateRegister: string
	countryOfProduce: string
	manufacturer: string
	responsible: string
	inventory: string
	yearOfIssue: number
	interVerificationInterval: number
	actOfEntering: string
	actOfEnteringId: string
	notes: string
	verificationDate: string
	nextVerificationDate: string
	certificate: string
	certificateId: string
	repair: string
	preservationDate: string
	dePreservationDate: string
	transferDate: string
	returnDate: string
	transferToDepartment: string
	writeOff: string
	person: string
	place: string
	lastPlace: string
	status: LocationStatus
}
export type LocationStatus = 'reserve' | 'used' | 'moved'

export interface ISiForm {
	instrument: IInstrumentDTO
	verification: IVerificationDTO
	location: ILocationDTO
}

export interface ISiDTO {
	instrument: IInstrumentDTO
	verification?: IVerificationDTO
}

export type Status = 'work' | 'repair' | 'decommissioning' | 'transferred'
export interface IGetSiDTO {
	section: string
	status?: Status
	all?: boolean
	page?: number
	size?: number
	sort?: ISort
	search?: ISearch
	filters?: IFilter[]
}

export interface IChangePositionDTO {
	sectionId: string
	newPosition: number
	oldPosition: number
}

export interface ISentSI {
	place: string
	si: ISI[]
	channel: string
	status: string
}

File: backend/internal/repository/postgres/pq_models/si.go
package pq_models

import (
	"time"

	"github.com/Alexander272/mersi/backend/internal/models"
)

type SI struct {
	models.SI
	RepairStart         time.Time `json:"repairStart" db:"repair_start"`
	RepairEnd           time.Time `json:"repairEnd" db:"repair_end"`
	RepairWork          string    `json:"repairWork" db:"repair_work"`
	NotificationChannel string    `json:"notification" db:"notification_channel"`
	BidType             string    `json:"bidType" db:"bid_type"`
}

func (s *SI) ToModel() *models.SI {
	return &models.SI{
		Id:                        s.Id,
		Position:                  s.Position,
		Name:                      s.Name,
		DateOfReceipt:             s.DateOfReceipt,
		Type:                      s.Type,
		FactoryNumber:             s.FactoryNumber,
		MeasurementLimits:         s.MeasurementLimits,
		Accuracy:                  s.Accuracy,
		StateRegister:             s.StateRegister,
		CountryOfProduce:          s.CountryOfProduce,
		Manufacturer:              s.Manufacturer,
		Responsible:               s.Responsible,
		Inventory:                 s.Inventory,
		YearOfIssue:               s.YearOfIssue,
		InterVerificationInterval: s.InterVerificationInterval,
		ActOfEntering:             s.ActOfEntering,
		ActOfEnteringId:           s.ActOfEnteringId,
		Notes:                     s.Notes,
		VerificationDate:          s.VerificationDate,
		NextVerificationDate:      s.NextVerificationDate,
		Certificate:               s.Certificate,
		CertificateId:             s.CertificateId,
		PreservationDate:          s.PreservationDate,
		DePreservationDate:        s.DePreservationDate,
		TransferDate:              s.TransferDate,
		ReturnDate:                s.ReturnDate,
		TransferToDepartment:      s.TransferToDepartment,
		WriteOff:                  s.WriteOff,
		Person:                    s.Person,
		Place:                     s.Place,
		LastPlace:                 s.LastPlace,
		Status:                    s.Status,
		Total:                     s.Total,
	}
}
